repos/fdbt-site/src/pages/api/createCaps.ts

```diff
@@ -11,105 +12,130 @@ import {
+        if (!(capType === 'fixedWeekdays' || capType === 'rollingDays')) {
+            errors.push({
+                id: 'fixed-weekdays',
+                errorMessage: 'Choose an option regarding your cap ticket start',
+            });
         }
 
-        caps.push({
-            name: trimmedCapName,
-            price: trimmedCapPrice,
-            durationAmount: trimmedCapDurationAmount,
-            durationUnits: cap.durationUnits as ExpiryUnit,
-        });
-    });
+        if (capType === 'fixedWeekdays') {
+            if (!isADayOfTheWeek(inputtedCap.startDay)) {
+                errors.push({
+                    id: 'start',
+                    errorMessage: 'Select a start day',
+                });
+            }
+        }
+
+        capStart = {
+            type: capType ? (capType as CapStart) : 'fixedWeekdays',
+            startDay: capType === 'rollingDays' ? undefined : (inputtedCap.startDay as DayOfTheWeek),
+        };
+    }
+
+    const cap = {
+        name: trimmedCapName,
+        price: trimmedCapPrice,
+        durationAmount: trimmedCapDurationAmount,
+        durationUnits: (inputtedCap.durationUnits as ExpiryUnit) || '',
+    };
+
+    const createCaps: CapInfo = {
+        cap,
+        capStart,
+    };
 
     return {
         errors,
-        caps,
+        createCaps,
     };
 };
 
-export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
     try {
-        const inputtedCaps: InputtedCap[] = [];
-        let i = 0;
-        const { cappedProductName } = req.body;
-        while (req.body[`capNameInput${i}`] !== undefined) {
-            const capName = req.body[`capNameInput${i}`] as string | undefined;
-            const capPrice = req.body[`capPriceInput${i}`] as string | undefined;
-            const capDurationAmount = req.body[`capDurationInput${i}`] as string | undefined;
-            const capDurationUnits = req.body[`capDurationUnitsInput${i}`] as string | undefined;
-
-            const inputtedCap: InputtedCap = {
-                name: capName,
-                price: capPrice,
-                durationAmount: capDurationAmount,
-                durationUnits: capDurationUnits,
-            };
-            inputtedCaps.push(inputtedCap);
-            i += 1;
-        }
+        const noc = getAndValidateNoc(req, res);
+        const { capName, capPrice, capDuration, capDurationUnits, capStart, startDay, id } = req.body;
+
+        const inputtedCap: InputtedCap = {
+            name: capName,
+            price: capPrice,
+            durationAmount: capDuration,
+            durationUnits: capDurationUnits,
+            type: capStart,
+            startDay: startDay,
+        };
 
-        const { caps, errors } = validateAndFormatCapInputs(inputtedCaps, cappedProductName);
+        const { createCaps, errors } = validateAndFormatCapInputs(inputtedCap);
```

Comment:
the function shouldn't return something called 'createCaps' - it should be createdCap